csharp/extractor/Semmle.Extraction.CSharp.DependencyFetching/DependencyManager.cs

```diff
@@ -214,37 +213,16 @@ private HashSet<string> AddFrameworkDlls(HashSet<string> dllPaths)
                         continue;
                     }
 
-                    if (useSubfolders)
-                    {
-                        dllPaths.Add(path);
-                        frameworkLocations.Add(path);
-                        continue;
-                    }
-
-                    try
-                    {
-                        var dlls = Directory.GetFiles(path, "*.dll", new EnumerationOptions { RecurseSubdirectories = false, MatchCasing = MatchCasing.CaseInsensitive });
-                        if (dlls.Length == 0)
-                        {
-                            logger.LogError($"No DLLs found in specified framework reference path '{path}'.");
-                            continue;
-                        }
-
-                        dllPaths.UnionWith(dlls);
-                        frameworkLocations.UnionWith(dlls);
-                    }
-                    catch (Exception e)
-                    {
-                        logger.LogError($"Error while searching for DLLs in '{path}': {e.Message}");
-                    }
+                    dllLocations.Add(new AssemblyLookupLocation(path, _ => true, useSubfolders));
+                    frameworkLocations.Add(path);
```

Comment:
@tamasvajk : Since the framework locations are only used as a prefix check later, then it should be fine that we still use the same prefix and not all the individual file names.